easypost/services/shipment_service.py

```diff
@@ -72,7 +72,7 @@ def get_next_page(
 
         return self.all(**params)
 
-    def regenerate_rates(self, id: str) -> Shipment:
+    def regenerate_rates(self, id: str) -> Dict[str, List[Rate]]:
```

Comment:
While this works because of duck typing, it would be better from a client perspective if the return type inherited from `EasyPostObject` since the method does return an `EasyPostObject` instance.

In my usage of this method, I pass the return value to `util.get_lowest_object_rate`, and with this `Dict` type it would fail to typecheck with pyright.

Reply:
Thanks for clarifying your use case! The `get_lowest_object_rate` function is intended for internal use by our other `get_lowest_rate` functions on a per-object basis, though it could certainly be used by end-users. The proposed return value is definitely the correct choice here; however, I can appreciate the desire to pass in the regenerated rates to the `lowest_rate` function. I've loosened the typing on `get_lowest_object_rate` to allow bare dictionaries in addition to EasyPostObjects which will retain the intention that this is for EasyPostObjects while allowing you to use it in the way you mentioned. We'll get this released in `v9.0.1`, please let us know if this doesn't fix the issue!